mapiproxy/servers/default/emsmdb/emsmdbp_provisioning.c

```diff
@@ -552,15 +550,25 @@ FolderId: 0x67ca828f02000001      Display Name: "                        ";  Con
 				mapistore_url = talloc_asprintf(mem_ctx, "%s0x%"PRIx64"/", fallback_url, current_fid);
 			}
 
-			/* Ensure the name is unique */
-			base_name = current_name;
-			j = 1;
-			while (openchangedb_get_fid_by_name(emsmdbp_ctx->oc_ctx, username, ipm_fid, current_name, &found_fid) == MAPI_E_SUCCESS) {
-				current_name = talloc_asprintf(mem_ctx, "%s (%d)", base_name, j);
-				j++;
+			/* According to [MS-OXOSFLD] Section 3.1.4.1, the special folder must be reused
+			   if one already exists by that name */
+			ret = openchangedb_get_fid_by_name(emsmdbp_ctx->oc_ctx, username, ipm_fid, current_name, &found_fid);
+			if (ret == MAPI_E_NOT_FOUND) {
+				mapistore_indexing_get_new_folderID_as_user(emsmdbp_ctx->mstore_ctx, username, &current_fid);
```

Comment:
I know that there are lot of calls in this function that are not checking return value for errors, but if we make some change I'd prefer to do it. Either `continue` or `goto error` (I'd prefer the latter)